src/dbus_api/pool.rs

```diff
@@ -360,48 +360,53 @@ fn get_pool_name(i: &mut IterAppend, p: &PropInfo<MTFn<TData>, TData>) -> Result
     get_pool_property(i, p, |(name, _, _)| Ok(name.to_owned()))
 }
 
-fn get_pool_total_physical_used(
-    i: &mut IterAppend,
-    p: &PropInfo<MTFn<TData>, TData>,
-) -> Result<(), MethodErr> {
-    fn get_used((_, uuid, pool): (Name, PoolUuid, &dyn Pool)) -> Result<String, MethodErr> {
-        let err_func = |_| {
-            MethodErr::failed(&format!(
-                "no total physical size computed for pool with uuid {}",
-                uuid
-            ))
-        };
-
-        pool.total_physical_used()
-            .map(|u| Ok(format!("{}", *u)))
-            .map_err(err_func)?
-    }
+fn get_all_properties(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
+    let message: &Message = m.msg;
+    let object_path = &m.path;
+    let dbus_context = m.tree.get_data();
+    let engine = dbus_context.engine.borrow();
 
-    get_pool_property(i, p, get_used)
-}
+    let default_return: HashMap<String, (bool, Variant<Box<dyn RefArg>>)> = HashMap::new();
+    let return_message = message.method_return();
 
-fn get_pool_total_physical_size(
-    i: &mut IterAppend,
-    p: &PropInfo<MTFn<TData>, TData>,
-) -> Result<(), MethodErr> {
-    get_pool_property(i, p, |(_, _, p)| {
-        Ok(format!("{}", *p.total_physical_size()))
-    })
-}
+    let uuid = get_data!(object_path; default_return; return_message).uuid;
+    let (_, pool) = get_pool!(engine; uuid; default_return; return_message);
 
-fn get_pool_state(i: &mut IterAppend, p: &PropInfo<MTFn<TData>, TData>) -> Result<(), MethodErr> {
-    get_pool_property(i, p, |(_, _, pool)| Ok(pool.state() as u16))
-}
+    let (tpu_success, total_physical_used) = match pool.total_physical_used() {
```

Comment:
We'll drop total physical used from the stats, as it is ill-defined.

Reply:
Oops! Sorry, misunderstood the issue you assigned me. I'll remove this.

Reply:
Resolved.